app/graph/types/user_type.rb

```diff
@@ -1,133 +1,134 @@
 UserType = GraphqlCrudOperations.define_default_type do
   name 'User'
-  description 'User type'
+  description 'A user of the application.'
 
   interfaces [NodeIdentification.interface]
 
-  field :dbid, types.Int
-  field :email, types.String
-  field :unconfirmed_email, types.String
-  field :providers, JsonStringType
-  field :uuid, types.String
-  field :profile_image, types.String
-  field :login, types.String
-  field :name, types.String
-  field :current_team_id, types.Int
-  field :permissions, types.String
-  field :jsonsettings, types.String
-  field :number_of_teams, types.Int
-  field :get_send_email_notifications, types.Boolean
-  field :get_send_successful_login_notifications, types.Boolean
-  field :get_send_failed_login_notifications, types.Boolean
-  field :bot_events, types.String
-  field :is_bot, types.Boolean
-  field :is_active, types.Boolean
-  field :two_factor, JsonStringType
-  field :settings, JsonStringType
-  field :accepted_terms, types.Boolean
-  field :last_accepted_terms_at, types.String
-  field :team_ids, types[types.Int]
-  field :user_teams, types.String
-
-  field :source_id do
-    type types.Int
+  field :email, types.String, 'Email'
+  field :unconfirmed_email, types.String, 'Email before confirmation'
+  field :providers, JsonStringType # TODO Document
+  field :uuid, types.String # TODO Document
+  field :profile_image, types.String, 'Picture' # TODO Rename to 'picture'
+  field :login, types.String, 'Login'
+  field :name, types.String, 'Name'
+  field :current_team_id, types.Int, 'Current team database id'
+  field :jsonsettings, types.String # TODO What's the difference with 'settings'?
+  field :number_of_teams, types.Int # TODO Remove because client can just count 'team_ids'?
+
+  # TODO Review notification settings to be extensible
+  field :get_send_email_notifications, types.Boolean, 'Setting for receiving email notifications about item activity'
+  field :get_send_successful_login_notifications, types.Boolean, 'Setting for receiving email notifications about all login activity'
+  field :get_send_failed_login_notifications, types.Boolean, 'Setting for receiving email notifications about failed login activity'
+
+  field :bot_events, types.String # TODO Why is this here?
+  field :is_bot, types.Boolean, 'Is this user a bot?'
+  field :is_active, types.Boolean, 'Is this user active?'
+  field :two_factor, JsonStringType # TODO Document
+  field :settings, JsonStringType, 'Settings' # TODO Show setting schema in description
+  field :accepted_terms, types.Boolean, 'Has use accepted latest terms?' # TODO Rename to 'has_accepted_latest_terms'
+  field :last_accepted_terms_at, types.String, 'Last date at which the user accepted the terms of use' # TODO Change to Int type
+  field :team_ids, types[types.Int], 'Teams where this user is a member (ids only)'
+  field :user_teams, types.String # TODO What's this in relation to above and to 'team_users'?
```

Comment:
Something I don't think we should keep... looking at the code, it returns the teams that the user is a member, as a hash where the key is the team slug and the value is the `TeamUser` object as a JSON